trie/triedb/pathdb/nodebufferlist.go

```diff
@@ -351,10 +529,23 @@ func (nf *nodebufferlist) getAllNodes() map[common.Hash]map[string]*trienode.Nod
 
 // getLayers return the size of cached difflayers.
 func (nf *nodebufferlist) getLayers() uint64 {
+	for {
+		if nf.isFlushing.Swap(true) {
```

Comment:
Does this change match the topic of this PR? It looks like an additional bug fix. If it is an unrelated fix, I suggest submitting it as a separate PR to facilitate tracking the source of each change.

Reply:
This bug is found when testing this pr and it's related.